common/build/eslint-config-fluid/scripts/print-configs.ts

```diff
@@ -16,12 +16,20 @@ import fs from "node:fs/promises";
 import path from "node:path";
 import { fileURLToPath } from "node:url";
 
-import { ESLint } from "eslint";
+import { loadESLint } from "eslint";
 import sortJson from "sort-json";
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 
+// Determine which config files to use based on ESLINT_USE_FLAT_CONFIG
+// NOTE: This is an ESLint limitation - loadESLint() only loads the correct ESLint class,
+// but we still need to manually determine which config file format to use.
+// Legacy ESLint cannot read flat config files (.mjs), and FlatESLint cannot read legacy config files (.js/.cjs).
+const useFlatConfig = process.env.ESLINT_USE_FLAT_CONFIG === "true";
+const configDir = useFlatConfig ? "../.eslint-print-configs" : "..";
+const configExt = useFlatConfig ? ".mjs" : ".js";
```

Comment:
This is probably worth a quick comment. Presumably we made the choice to be a bit more explicit with our file extensions for the new configs?